src/frontend/js/pages/TeacherDashboardContractsLoader/TeacherDashboardCourseContractsLoader/index.tsx

```diff
@@ -1,50 +1,26 @@
 import { FormattedMessage, defineMessages } from 'react-intl';
-import { useParams } from 'react-router-dom';
-import queryString from 'query-string';
 
-import { useState } from 'react';
 import { TeacherDashboardCourseSidebar } from 'widgets/Dashboard/components/TeacherDashboardCourseSidebar';
 import { DashboardLayout } from 'widgets/Dashboard/components/DashboardLayout';
-import { location } from 'utils/indirection/window';
 import TeacherDashboardContracts from '../TeacherDashboardContracts';
-import ContractFilters, { ContractListFilters } from '../ContractFilters';
 
 const messages = defineMessages({
   pageTitle: {
     defaultMessage: 'Contracts',
     description: "Use for the page title of the course's contracts page",
     id: 'pages.TeacherDashboardCourseContractsLoader.pageTitle',
   },
-  loading: {
-    defaultMessage: 'Loading contracts...',
-    description: 'Message displayed while loading contracts',
-    id: 'pages.TeacherDashboardCourseContractsLoader.loading',
-  },
 });
 
 export const TeacherDashboardCourseContractsLoader = () => {
-  const { courseId } = useParams<{ courseId?: string }>();
-  const { page = '1' }: { page?: string } = queryString.parse(location.search);
-  const [filters, setFilters] = useState<ContractListFilters>({});
-  const onFiltersChange = (newFilters: ContractListFilters) => {
-    setFilters(newFilters);
-  };
```

Comment:
The idea of these `*Loader` is to handle outside world interaction that will initialize the page.
It handle the initial loading.
If you remove this logic from here, you can remove the component.

Reply:
Indeed, the term Loader is no more relevant for this kind of component, I should replace Loader by Layout.